coalib/coala.py

```diff
@@ -61,25 +68,19 @@ def main():
 
             return 0
 
-    except BaseException as exception:  # pylint: disable=broad-except
-        return get_exitcode(exception, log_printer)
-
-    import functools
+        if args.ci:
+            exitcode = mode_ci(console_printer, args)
+            return exitcode
 
-    from coalib.coala_main import run_coala
+        if args.format:
+            exitcode = mode_format()
+            return exitcode
 
-    partial_print_sec_beg = functools.partial(
-        print_section_beginning,
-        console_printer)
-    results, exitcode, _ = run_coala(
-        print_results=print_results,
-        acquire_settings=acquire_settings,
-        print_section_beginning=partial_print_sec_beg,
-        nothing_done=nothing_done,
-        console_printer=console_printer)
+    except BaseException as exception:  # pylint: disable=broad-except
+        return get_exitcode(exception, log_printer)
 
+    exitcode = mode_normal(console_printer, log_printer)
```

Comment:
well the try except was needed for the showing bears but not for any of the modes right? ;)